### PR TITLE
Use Rails 4 syntax for grouped counts

### LINKED ISSUE #3984: Stats broken in admin interface
https://www.whatdotheyknow.com/admin/stats

ss 2017-05-13 at 12 35 10

### PR DESCRIPTION
**Pull request by lizconlan**
Fixes #3984

### CODE REVIEW

**Comment by lizconlan** [comment]
Should this be a hotfix if we're about to upgrade Belgium to 0.28?

**Comment by lizconlan** [comment]
ToDo: change commit comment to indicate Rails 4.1 rather than Rails 4 syntax

File: doc/CHANGES.md [hunk 1]
```diff
@@ -65,6 +65,8 @@
   are no requests (Liz Conlan)
 * Switch to Trusty as the preferred OS for Travis CI and use Debian Wheezy as
   the new Vagrant default (Liz Conlan)
+* Fix bug which prevented stats from being displayed in the admin interface
```
Comment (garethrees) [inline doc/CHANGES.md]: I'd say we don't need a changelog entry as there wasn't a release introducing the bug

File: spec/controllers/admin_general_controller_spec.rb [hunk 2]
```diff
@@ -150,4 +150,66 @@
...
+
+      8.times { FactoryGirl.create(:successful_request) }
+      2.times { FactoryGirl.create(:info_request) }
+      FactoryGirl.create(:attention_requested_request)
+
+      get :stats
+      expect(assigns[:request_by_state]).to be_a(Hash)
+      expect(assigns[:request_by_state]).to include("successful" => 8)
+      expect(assigns[:request_by_state]).to include("waiting_response" => 2)
+      expect(assigns[:request_by_state]).to include("attention_requested" => 1)
```
Comment (garethrees) [inline spec/controllers/admin_general_controller_spec.rb]: +1 to the extra tests!

Any reason this isn't a [single expect](http://www.betterspecs.org/#single), e.g:

```ruby
expect(assigns[:request_by_state]).
  to eq({ 'successful' => 8,
          'waiting_response' => 2,
          'attention_requested' => 1 })
```

Ditto for the spec below?

Comment (lizconlan) [inline spec/controllers/admin_general_controller_spec.rb]: only that I wasn't entirely sure what the order would be

Comment (garethrees) [inline spec/controllers/admin_general_controller_spec.rb]: Ah! Don't think it matters though:

```ruby
{ :a => 'b', :c => 3 } == { :c => 3, :a => 'b' }
# => true
```

Comment (lizconlan) [inline spec/controllers/admin_general_controller_spec.rb]: \o/

**Pull request merged**